iceoryx2/src/config.rs

```diff
@@ -403,29 +485,41 @@ impl Config {
+    /// load a default config by looking it up in the system. First it checks if a project local config file
+    /// exists, then if a config file in the user directory exist and then if a global config file exist. If
     /// [`Config::setup_global_config_from_file()`]
     /// is called after this function was called, no file will be loaded since the global default
     /// config was already populated.
     pub fn global_config() -> &'static Config {
+        let origin = "Config::global_config()";
         if !ICEORYX2_CONFIG.is_initialized() {
-            match Config::setup_global_config_from_file(unsafe {
-                &FilePath::new_unchecked(DEFAULT_CONFIG_FILE)
-            }) {
-                Ok(_) => (),
-                Err(ConfigCreationError::FailedToOpenConfigFile) => {
-                    debug!(from "Config::global_config()", "Default config file not found, populate config with default values.");
-                    ICEORYX2_CONFIG.set_value(Config::default());
-                }
-                Err(ConfigCreationError::FailedToReadConfigFileContents) => {
-                    warn!(from "Config::global_config()", "Default config file found but unable to read content, populate config with default values.");
-                    ICEORYX2_CONFIG.set_value(Config::default());
-                }
-                Err(ConfigCreationError::UnableToDeserializeContents) => {
-                    warn!(from "Config::global_config()", "Default config file found but unable to load data, populate config with default values.");
-                    ICEORYX2_CONFIG.set_value(Config::default());
+            let mut is_config_file_set = false;
+            if let Err(e) = Self::iterate_over_config_files(|config_file_path| {
+                match Config::setup_global_config_from_file(&config_file_path) {
+                    Ok(_) => {
+                        is_config_file_set = true;
+                        CallbackProgression::Stop
+                    }
+                    Err(ConfigCreationError::FailedToOpenConfigFile) => {
+                        CallbackProgression::Continue
+                    }
+                    Err(e) => {
+                        warn!(from origin,
+                                "Config file found \"{}\" but a failure occurred ({:?}) while reading the content.",
+                                config_file_path, e);
+                        CallbackProgression::Continue
+                    }
                 }
+            }) {
+                warn!(from origin,
+                    "A failure occurred ({:?}) while looking up all available config files.", e);
+            }
+
+            if !is_config_file_set {
+                warn!(from origin,
+                    "Since no config file could be loaded, a config with internal default values will be set.");
+                ICEORYX2_CONFIG.set_value(Config::default());
```

Comment:
I would do this only if no config file was found, not when it's available but corrupt.